docs/ref/models/fields.txt

```diff
@@ -1547,6 +1547,38 @@ The possible values for :attr:`~ForeignKey.on_delete` are found in
     integrity, this will cause an :exc:`~django.db.IntegrityError` unless
     you manually add an SQL ``ON DELETE`` constraint to the database field.
 
+* .. attribute:: DB_CASCADE
+
+    .. versionadded:: 3.1
+
+    Django takes no direct action, behaves exactly like DO_NOTHING,
+    but also sets ON DELETE CASCADE as a SQL constraint. The cascade
+    deletion then happens at the database level.
+
+* .. attribute:: DB_NO_ACTION
+
+    .. versionadded:: 3.1
+
+    Django takes no direct action, behaves exactly like DO_NOTHING, but
+    also explicitly sets ON DELETE NO ACTION as a SQL constraint.
```

Comment:
Is there any reason to add `DB_NO_ACTION`? As far as I'm concerned that's the default behavior in all databases, and it behaves the same as `DO_NOTHING`.

Reply:
Originally, the whole PR was just DB_CASCADE, but then we expanded this into other DB_* actions, and I guess DB_NO_ACTION was simply added for completeness. I agree though that it's probably not needed. I can remove it. Fewer new pieces :).

Reply:
If you don't have `DB_NO_ACTION`, how do you revert to this in a migration if you have previously set `DB_CASCADE` (or one of the others)? I think it is required...

The whole point of adding this was to avoid changing the behaviour of `DO_NOTHING` which changes nothing in the database.

Reply:
@pope1ni after looking at this again, I agree that DO_NOTHING should be as unchanged as possible. I've made that correction in the latest commits.

 However, I believe one could revert DB_CASCADE by setting on_delete to whatever else (e.g. DO_NOTHING). The model would serialize different table SQL and it would get migrated. Please correct me if this is not correct.